metadata-ingestion/src/datahub/ingestion/source/schema_inference/json.py

```diff
@@ -29,9 +29,24 @@
 
 
 class JsonInferrer(SchemaInferenceBase):
-    def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
+    def merge_new_dictionary(self, complete_dict, new_dict):
+        for key in new_dict.keys():
+            if key not in complete_dict.keys():
+                complete_dict[key] = new_dict[key]
+
+        return complete_dict
+
 
-        datastore = ujson.load(file)
+    def parse_json(self, file):
+        complete_datastore = {}
+        for line in file:
+            datastore = ujson.loads(line.decode('utf8'))
```

Comment:
If the `for` loop is already iterating over lines, then you should be getting strings. But you are getting a byte stream in `line`?